controllers/openstackclient_controller.go

```diff
@@ -93,26 +93,76 @@ func (r *OpenStackClientReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 	envVars := make(map[string]common.EnvSetter)
 
 	// check for required secrets
-	hashes := []ospdirectorv1beta1.Hash{}
-	sshSecret, secretHash, err := common.GetSecret(r, instance.Spec.DeploymentSSHSecret, instance.Namespace)
+	_, _, err = common.GetSecret(r, instance.Spec.DeploymentSSHSecret, instance.Namespace)
 	if err != nil && errors.IsNotFound(err) {
 		return ctrl.Result{RequeueAfter: time.Second * 20}, fmt.Errorf("DeploymentSSHSecret secret does not exist: %v", err)
 	} else if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	hashes = append(hashes, ospdirectorv1beta1.Hash{Name: sshSecret.Name, Hash: secretHash})
+	templateParameters := make(map[string]interface{})
+	cmLabels := common.GetLabels(instance.Name, openstackclient.AppLabel)
+	cms := []common.Template{
+		// Custom CM holding Tripleo deployment environment parameter files
+		{
+			Name:      "tripleo-deploy-config-custom",
+			Namespace: instance.Namespace,
+			Type:      common.TemplateTypeCustom,
+			Labels:    cmLabels,
+		},
+		// Custom CM holding Tripleo net-config files then used in parameter files
+		{
+			Name:      "tripleo-net-config",
+			Namespace: instance.Namespace,
+			Type:      common.TemplateTypeCustom,
+			Labels:    cmLabels,
+		},
+	}
+	err = common.EnsureConfigMaps(r, instance, cms, &envVars)
+	if err != nil {
+		return ctrl.Result{}, nil
+	}
 
-	// check for required configMaps
-	configMaps := []string{
-		"tripleo-deploy-config",
-		"tripleo-deploy-sh",
+	// get tripleo-deploy-config, created/rendered by overcloudipset controller
```

Comment:
Maybe should say `tripleo-deploy-config-custom` instead

Reply:
yes, indeed

Reply:
done